llama_stack/providers/inline/tool_runtime/brave_search/brave_search.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from typing import Any, Dict, List
+
+import requests
+
+from llama_stack.apis.tools import InvokeToolResult, Tool, ToolRuntime
+from llama_stack.providers.datatypes import ToolsProtocolPrivate
+
+from .config import BraveSearchToolConfig
+
+
+class BraveSearchToolRuntimeImpl(ToolsProtocolPrivate, ToolRuntime):
```

Comment:
you can now add `NeedsProviderData` mixin here and request the API_KEY through the provider_data mechanism we have. Look at the Together inference impl.

Reply:
done.